### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Husain01**
Added 
- Auth Context
- Auth Reducer
- Requires Auth Check to 
    - Login
    - SignUp
- Add Logout button
Link: https://wachbuch-efl4tjobh-husain01.vercel.app/

### CODE REVIEW

File: src/Reducers/Auth/AuthReducer.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        user: action.payload.user,
+        token: action.payload.token,
+      };
+    case "LOGOUT":
+      return {
+        ...state,
+        user: null,
+        token: null,
+      };
+    case "CHECKUSER":
```
Comment (ApurvaSawant11) [inline src/Reducers/Auth/AuthReducer.js]: Here, three of your cases have the same return object. You can have a general case like `SET_USER` for `LOGIN`, `SIGNUP` and `CHECKUSER`.

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -1,52 +1,61 @@
 import React from "react";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import logo from "../../assets/logo-sm.png";
-import logo_txt from '../../assets/logo-txt.png'
+import logo_txt from "../../assets/logo-txt.png";
```
Comment (ApurvaSawant11) [inline src/components/Navbar/Navbar.js:4]: Try barrel imports here. 

Check this https://srinivasankk.com/javascript-barrel/  or if it's still not clear, you can check one of my projects

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me! Just gave a few suggestions

File: src/Reducers/Auth/AuthReducer.js (hunk 1 above)
Comment (junaidshaikh-js) [reply src/Reducers/Auth/AuthReducer.js]: Or you can combine multiple cases together.

```js
switch(action.type) {
 case 'LOGIN':
 case 'SIGNUP': 
  return {
    ....
   }
}
```

File: src/App.js [hunk 3]
```diff
@@ -1,25 +1,69 @@
...
-       <Route path="/home" element={<Home/>}/> 
-       <Route path="/video" element={<SingleVideo/>}/> 
-       <Route path="/login" element={<Login/>}/> 
-       <Route path="/signup" element={<Signup/>}/> 
-       <Route path="/playlist" element={<Playlist/>}/> 
-       <Route path="/liked" element={<Liked/>}/> 
-       <Route path="/watchlater" element={<WatchLater/>}/> 
-       <Route path="/history" element={<History/>}/> 
+        <Route path="/" element={<Home />} />
+        <Route path="/home" element={<Home />} />
```
Comment (junaidshaikh-js) [inline src/App.js:27]: Why two paths for the same component?

File: src/pages/Auth/Login.js [hunk 4]
```diff
@@ -1,8 +1,68 @@
...
+    e.preventDefault();
+    setUser(guestUser);
+  };
+
+  const loginHandler = async (e) => {
+    e.preventDefault();
+    if (user.email !== "" && user.password !== "") {
+      try {
+        const res = await loginService(user);
+        console.log(res)
```
Comment (junaidshaikh-js) [inline src/pages/Auth/Login.js]: remove console logs

**Review by junaidshaikh-js — COMMENTED** [review]
Good work done. 

Suggested some changes, everything else looks fine.

**Pull request merged**